todo/views.py

```diff
@@ -1,6 +1,77 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
 
-from django.shortcuts import render
+from django.contrib.auth.forms import UserCreationForm
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.http import HttpResponseRedirect
+from django.shortcuts import redirect
+from django.urls import reverse_lazy
+from django.views.generic import ListView, DetailView, FormView, DeleteView, CreateView
+
+from .forms import TaskForm
+from .models import Task
+
 
 # Create your views here.
+
+class AbstractLogin(LoginRequiredMixin):
+    login_url = '/login/'
+    redirect_field_name = reverse_lazy('tasklist')
+
+    class Meta:
+        abstract = True
```

Comment:
abstract View ? What is that ? abstract=True is used in models, when you would like to avoid creating table in db